arch/arm/core/aarch32/cortex_m/reset.S

```diff
@@ -71,6 +71,13 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
     isb
 #endif
 
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SPLIM)
```

Comment:
Can you put this inside the CONFIG_INIT_ARCH_HW_AT_BOOT, so together with the CONTROL register above?
Otherwise this is not needed since SPLIM regs are 0 when we come out of warm reset.